### ISSUE TITLE
Debugger Issue: Cannot start dotnet

### ISSUE THREAD

**Issue by jxs55km** [dotnet/vscode-csharp#6365]
## Environment data
`dotnet --info` output:
.NET SDK:
 Version:   7.0.401
 Commit:    eb26aacfec

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.13
 OS Platform: Darwin
 RID:         osx.10.13-x64
 Base Path:   /usr/local/share/dotnet/sdk/7.0.401/

Host:
  Version:      7.0.11
  Architecture: x64
  Commit:       ecb34f85ec

.NET SDKs installed:
  7.0.401 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 7.0.11 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 7.0.11 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download

**VSCode version**: 1.82.2
**C# Extension**: 2.1.2
**Using OmniSharp**: false

## Steps to reproduce

1. Create new folder
2. Create new file
3. Select run and debug

## Expected  behavior
The code to run, and for the debug console to read "Hello World!"

## Actual behavior
Debug console output: The program '[1524] VS Files.dll' has exited with code -1 (0xffffffff).

Error message: Unable to start program '/usr/local/share/dotnet/dotnet'. Unknown Error: 0x92330062

## Additional context
I am very new to coding, and am struggling to get my debugger setup on my Mac.

I can run the basic "Hello World" program through VS Code's built-in terminal, OSX terminal, and with "run without debugging".

I have rebooted VS and my machine, uninstalled/reinstalled VS Code and .NET, as well as searched for many hours trying to find another person with a similar issue.

I lack a lot of prerequisite knowledge of how all of this works, so I apologize in advance if the solution to this is seemingly obvious.

**Comment by gregg-miskelly** [comment]
Very strange. That error is internal error code `E_DBGSHIM_INIT_ERROR`, which happens if we couldn't load libdbgshim.dylib, or it was missing one of its expected exports. I don't have any great guesses as to why that would be on your computer.

**Comment by gregg-miskelly** [comment]
Here is how you could try to trouble shoot this:

1. Open a terminal
2. `cd` to the `<csharp-extension-root>\.debugger\<processor-architecture>` directory. Probably: `cd ~/.vscode/extensions/ms-dotnettools.csharp-2.0.283-darwin-x86_64/.debugger/x86_64`
3. Enable diagnostics from `dyld`. You probably want the below list.
4. Run: `./vsdbg-ui --server`
5. If you don't have a launch.json file, run the ".NET: Generate Assets for Build and Debug" command from VS Code's command pallet
6. Open your launch.json, and add `"debugServer": 4711`
7. Try to debug
8. Look for 'dbgshim' in the output spew from your terminal, and paste it here

Environment variables to enable diagnostics:
```
export DYLD_PRINT_LIBRARIES=1
export DYLD_PRINT_SEARCHING=1
export DYLD_PRINT_WARNINGS=1
```

Example logging produced from loading dbgshim --
```
dyld[52487]: find path "/Users/greggmiskelly/.vscode-server/extensions/ms-dotnettools.csharp-2.0.283-darwin-arm64/.debugger/arm64/libdbgshim.dylib"
dyld[52487]:   possible path(original path on disk): "/Users/greggmiskelly/.vscode-server/extensions/ms-dotnettools.csharp-2.0.283-darwin-arm64/.debugger/arm64/libdbgshim.dylib"
dyld[52487]:   found: dylib-from-disk: "/Users/greggmiskelly/.vscode-server/extensions/ms-dotnettools.csharp-2.0.283-darwin-arm64/.debugger/arm64/libdbgshim.dylib"
dyld[52487]: <96780707-C89C-35E0-9E15-D179BBF8EB4C> /Users/greggmiskelly/.vscode-server/extensions/ms-dotnettools.csharp-2.0.283-darwin-arm64/.debugger/arm64/libdbgshim.dylib
```
```
```

**Comment by vike2000** [comment]
# Similar issue



(btw: `>CSharp: report an issue` doesn't prefill but just opens https://github.com/dotnet/vscode-csharp.git in my browser, Chrome.)

## Environment data
`dotnet --info` output:

``` text
.NET SDK (reflecting any global.json):
 Version:   6.0.106
 Commit:    e83a22f558

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.13
 OS Platform: Darwin
 RID:         osx.10.13-x64
 Base Path:   /usr/local/share/dotnet/sdk/6.0.106/

Host:
  Version:      7.0.0-preview.7.22375.6
  Architecture: x64
  Commit:       eecb028078

.NET SDKs installed:
  3.1.420 [/usr/local/share/dotnet/sdk]
  5.0.408 [/usr/local/share/dotnet/sdk]
  6.0.100-rc.2.21505.57 [/usr/local/share/dotnet/sdk]
  6.0.100 [/usr/local/share/dotnet/sdk]
  6.0.106 [/usr/local/share/dotnet/sdk]
  7.0.100-preview.7.22377.5 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.26 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.17 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.0-rc.2.21480.10 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.0 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.6 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.0-preview.7.22376.6 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.26 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.17 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.0-rc.2.21480.5 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.0 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.6 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0-preview.7.22375.6 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  Not set

global.json file:
  /Users/vike/…/repo/global.json
```

**VS Code version**: 1.81.1, same behaviour after upgrading to 1.82.2
**C# Extension version**: 2.4.4

**Using OmniSharp**: yes

## OmniSharp log

How to procure?

## Steps to reproduce

1. Using `launch.json` configurations entry:
  ```
			{		"name"					:	"Launch (coreclr)"
//			,		"debugServer"			:	 4711
			,		"request"				:	"launch"
			,		"type"					:	"coreclr"
			,		"preLaunchTask"			:	"dotnet build"
			,		"cwd"					:	"${workspaceFolder}"
			,		"program"				:	"/usr/local/share/godot/godot"
//			,		"program"				:	"C:\\usr\\local\\share\\godot\\Godot_v4.1.1-stable_mono_win64.exe"
			// See which arguments are available here:
			// https://docs.godotengine.org/en/stable/getting_started/editor/command_line_tutorial.html
			,		"args"
			:	[	"--path"				,	"${workspaceFolder}/project.godot"
				,	"--windowed"
				,	"--position"			,	"32,64"
				,	"--resolution"			,	"1280x720"
				,	"--rendering-driver"	,	"opengl3"
				]}
```

## Expected and actual  behaviors

Very similar to OP but with program `/usr/local/share/godot/godot` instead of `dotnet`.

**Debug console**: The program '[33824] godot' has exited with code -1 (0xffffffff).

**Error notification**: Unable to start program '/usr/local/share/godot/godot'. Unknown Error: 0x92330062

## Additional context

Re `./vsdbg-ui --server` per @gregg-miskelly's https://github.com/dotnet/vscode-csharp/issues/6365#issuecomment-1720256692

Excerpt:
``` text
…
Waiting for communication on port 4711...
>> accepted connection from client
dyld: loaded: /Users/vike/.vscode/extensions/ms-dotnettools.csharp-2.4.4-darwin-x64/.debugger/x86_64/libvsdbg.dylib
dyld: loaded: /Users/vike/.vscode/extensions/ms-dotnettools.csharp-2.4.4-darwin-x64/.debugger/x86_64/libvsdebugeng.impl.dylib
dyld: loaded: /Users/vike/.vscode/extensions/ms-dotnettools.csharp-2.4.4-darwin-x64/.debugger/x86_64/libdbgshim.dylib
>> client connection closed
```



<details>

<summary>Full output</summary>

``` text
dyld: warning, unknown environment variable: DYLD_PRINT_SEARCHING
dyld: loaded: /Users/vike/.vscode/extensions/ms-dotnettools.csharp-2.4.4-darwin-x64/.debugger/x86_64/./vsdbg-ui
dyld: loaded: /Users/vike/.vscode/extensions/ms-dotnettools.csharp-2.4.4-darwin-x64/.debugger/x86_64/libvsdebugeng.dylib
dyld: loaded: /usr/lib/libc++.1.dylib
dyld: loaded: /Users/vike/.vscode/extensions/ms-dotnettools.csharp-2.4.4-darwin-x64/.debugger/x86_64/libvsbaseservices.dylib
dyld: loaded: /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation
dyld: loaded: /System/Library/Frameworks/CoreServices.framework/Versions/A/CoreServices
dyld: loaded: /System/Library/Frameworks/Security.framework/Versions/A/Security
dyld: loaded: /usr/lib/libSystem.B.dylib
dyld: loaded: /usr/lib/libc++abi.dylib
dyld: loaded: /usr/lib/system/libcache.dylib
dyld: loaded: /usr/lib/system/libcommonCrypto.dylib
dyld: loaded: /usr/lib/system/libcompiler_rt.dylib
dyld: loaded: /usr/lib/system/libcopyfile.dylib
dyld: loaded: /usr/lib/system/libcorecrypto.dylib
dyld: loaded: /usr/lib/system/libdispatch.dylib
dyld: loaded: /usr/lib/system/libdyld.dylib
dyld: loaded: /usr/lib/system/libkeymgr.dylib
dyld: loaded: /usr/lib/system/liblaunch.dylib
dyld: loaded: /usr/lib/system/libmacho.dylib
dyld: loaded: /usr/lib/system/libquarantine.dylib
dyld: loaded: /usr/lib/system/libremovefile.dylib
dyld: loaded: /usr/lib/system/libsystem_asl.dylib
dyld: loaded: /usr/lib/system/libsystem_blocks.dylib
dyld: loaded: /usr/lib/system/libsystem_c.dylib
dyld: loaded: /usr/lib/system/libsystem_configuration.dylib
dyld: loaded: /usr/lib/system/libsystem_coreservices.dylib
dyld: loaded: /usr/lib/system/libsystem_darwin.dylib
dyld: loaded: /usr/lib/system/libsystem_dnssd.dylib
dyld: loaded: /usr/lib/system/libsystem_info.dylib
dyld: loaded: /usr/lib/system/libsystem_m.dylib
dyld: loaded: /usr/lib/system/libsystem_malloc.dylib
dyld: loaded: /usr/lib/system/libsystem_network.dylib
dyld: loaded: /usr/lib/system/libsystem_networkextension.dylib
dyld: loaded: /usr/lib/system/libsystem_notify.dylib
dyld: loaded: /usr/lib/system/libsystem_sandbox.dylib
dyld: loaded: /usr/lib/system/libsystem_secinit.dylib
dyld: loaded: /usr/lib/system/libsystem_kernel.dylib
dyld: loaded: /usr/lib/system/libsystem_platform.dylib
dyld: loaded: /usr/lib/system/libsystem_pthread.dylib
dyld: loaded: /usr/lib/system/libsystem_symptoms.dylib
dyld: loaded: /usr/lib/system/libsystem_trace.dylib
dyld: loaded: /usr/lib/system/libunwind.dylib
dyld: loaded: /usr/lib/system/libxpc.dylib
dyld: loaded: /usr/lib/closure/libclosured.dylib
dyld: loaded: /usr/lib/libobjc.A.dylib
dyld: loaded: /usr/lib/libDiagnosticMessagesClient.dylib
dyld: loaded: /usr/lib/libicucore.A.dylib
dyld: loaded: /usr/lib/libz.1.dylib
dyld: loaded: /System/Library/Frameworks/CFNetwork.framework/Versions/A/CFNetwork
dyld: loaded: /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/FSEvents.framework/Versions/A/FSEvents
dyld: loaded: /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/CarbonCore.framework/Versions/A/CarbonCore
dyld: loaded: /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/Metadata.framework/Versions/A/Metadata
dyld: loaded: /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/OSServices.framework/Versions/A/OSServices
dyld: loaded: /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/SearchKit.framework/Versions/A/SearchKit
dyld: loaded: /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/AE.framework/Versions/A/AE
dyld: loaded: /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/LaunchServices.framework/Versions/A/LaunchServices
dyld: loaded: /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/DictionaryServices.framework/Versions/A/DictionaryServices
dyld: loaded: /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/SharedFileList.framework/Versions/A/SharedFileList
dyld: loaded: /usr/lib/libauto.dylib
dyld: loaded: /usr/lib/libbsm.0.dylib
dyld: loaded: /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit
dyld: loaded: /System/Library/Frameworks/SystemConfiguration.framework/Versions/A/SystemConfiguration
dyld: loaded: /usr/lib/libsqlite3.dylib
dyld: loaded: /usr/lib/libxml2.2.dylib
dyld: loaded: /usr/lib/libnetwork.dylib
dyld: loaded: /usr/lib/libapple_nghttp2.dylib
dyld: loaded: /usr/lib/libenergytrace.dylib
dyld: loaded: /usr/lib/system/libkxld.dylib
dyld: loaded: /usr/lib/libpcap.A.dylib
dyld: loaded: /usr/lib/libboringssl.dylib
dyld: loaded: /usr/lib/libusrtcp.dylib
dyld: loaded: /usr/lib/libcoretls.dylib
dyld: loaded: /usr/lib/libcoretls_cfhelpers.dylib
dyld: loaded: /usr/lib/libapple_crypto.dylib
dyld: loaded: /System/Library/Frameworks/Foundation.framework/Versions/C/Foundation
dyld: loaded: /usr/lib/libOpenScriptingUtil.dylib
dyld: loaded: /usr/lib/libpam.2.dylib
dyld: loaded: /usr/lib/libxar.1.dylib
dyld: loaded: /System/Library/Frameworks/DiskArbitration.framework/Versions/A/DiskArbitration
dyld: loaded: /usr/lib/libarchive.2.dylib
dyld: loaded: /usr/lib/liblangid.dylib
dyld: loaded: /usr/lib/libCRFSuite.dylib
dyld: loaded: /usr/lib/libbz2.1.0.dylib
dyld: loaded: /usr/lib/liblzma.5.dylib
dyld: loaded: /System/Library/Frameworks/NetFS.framework/Versions/A/NetFS
dyld: loaded: /System/Library/PrivateFrameworks/NetAuth.framework/Versions/A/NetAuth
dyld: loaded: /System/Library/PrivateFrameworks/login.framework/Versions/A/Frameworks/loginsupport.framework/Versions/A/loginsupport
dyld: loaded: /System/Library/PrivateFrameworks/TCC.framework/Versions/A/TCC
dyld: loaded: /usr/lib/libmecabra.dylib
dyld: loaded: /System/Library/Frameworks/ApplicationServices.framework/Versions/A/ApplicationServices
dyld: loaded: /System/Library/Frameworks/CoreGraphics.framework/Versions/A/CoreGraphics
dyld: loaded: /System/Library/Frameworks/CoreText.framework/Versions/A/CoreText
dyld: loaded: /System/Library/Frameworks/ImageIO.framework/Versions/A/ImageIO
dyld: loaded: /System/Library/Frameworks/ColorSync.framework/Versions/A/ColorSync
dyld: loaded: /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/ATS.framework/Versions/A/ATS
dyld: loaded: /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/ColorSyncLegacy.framework/Versions/A/ColorSyncLegacy
dyld: loaded: /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/HIServices.framework/Versions/A/HIServices
dyld: loaded: /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/LangAnalysis.framework/Versions/A/LangAnalysis
dyld: loaded: /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/PrintCore.framework/Versions/A/PrintCore
dyld: loaded: /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/QD.framework/Versions/A/QD
dyld: loaded: /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/SpeechSynthesis.framework/Versions/A/SpeechSynthesis
dyld: loaded: /System/Library/PrivateFrameworks/SkyLight.framework/Versions/A/SkyLight
dyld: loaded: /System/Library/Frameworks/Accelerate.framework/Versions/A/Accelerate
dyld: loaded: /System/Library/Frameworks/CoreDisplay.framework/Versions/A/CoreDisplay
dyld: loaded: /System/Library/Frameworks/IOSurface.framework/Versions/A/IOSurface
dyld: loaded: /System/Library/Frameworks/Metal.framework/Versions/A/Metal
dyld: loaded: /System/Library/PrivateFrameworks/MultitouchSupport.framework/Versions/A/MultitouchSupport
dyld: loaded: /System/Library/Frameworks/QuartzCore.framework/Versions/A/QuartzCore
dyld: loaded: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vImage.framework/Versions/A/vImage
dyld: loaded: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/vecLib
dyld: loaded: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libvDSP.dylib
dyld: loaded: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBNNS.dylib
dyld: loaded: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libQuadrature.dylib
dyld: loaded: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libvMisc.dylib
dyld: loaded: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libLAPACK.dylib
dyld: loaded: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
dyld: loaded: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libLinearAlgebra.dylib
dyld: loaded: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libSparse.dylib
dyld: loaded: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libSparseBLAS.dylib
dyld: loaded: /System/Library/PrivateFrameworks/GPUWrangler.framework/Versions/A/GPUWrangler
dyld: loaded: /System/Library/PrivateFrameworks/IOAccelerator.framework/Versions/A/IOAccelerator
dyld: loaded: /System/Library/PrivateFrameworks/IOPresentment.framework/Versions/A/IOPresentment
dyld: loaded: /System/Library/PrivateFrameworks/DSExternalDisplay.framework/Versions/A/DSExternalDisplay
dyld: loaded: /System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libCoreFSCache.dylib
dyld: loaded: /System/Library/Frameworks/CoreImage.framework/Versions/A/CoreImage
dyld: loaded: /System/Library/Frameworks/CoreVideo.framework/Versions/A/CoreVideo
dyld: loaded: /System/Library/Frameworks/OpenGL.framework/Versions/A/OpenGL
dyld: loaded: /System/Library/PrivateFrameworks/GraphVisualizer.framework/Versions/A/GraphVisualizer
dyld: loaded: /System/Library/Frameworks/MetalPerformanceShaders.framework/Versions/A/MetalPerformanceShaders
dyld: loaded: /usr/lib/libFosl_dynamic.dylib
dyld: loaded: /System/Library/PrivateFrameworks/FaceCore.framework/Versions/A/FaceCore
dyld: loaded: /System/Library/Frameworks/OpenCL.framework/Versions/A/OpenCL
dyld: loaded: /usr/lib/libcompression.dylib
dyld: loaded: /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/ATS.framework/Versions/A/Resources/libFontParser.dylib
dyld: loaded: /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/ATS.framework/Versions/A/Resources/libFontRegistry.dylib
dyld: loaded: /System/Library/Frameworks/ImageIO.framework/Versions/A/Resources/libJPEG.dylib
dyld: loaded: /System/Library/Frameworks/ImageIO.framework/Versions/A/Resources/libTIFF.dylib
dyld: loaded: /System/Library/Frameworks/ImageIO.framework/Versions/A/Resources/libPng.dylib
dyld: loaded: /System/Library/Frameworks/ImageIO.framework/Versions/A/Resources/libGIF.dylib
dyld: loaded: /System/Library/Frameworks/ImageIO.framework/Versions/A/Resources/libJP2.dylib
dyld: loaded: /System/Library/Frameworks/ImageIO.framework/Versions/A/Resources/libRadiance.dylib
dyld: loaded: /System/Library/PrivateFrameworks/AppleJPEG.framework/Versions/A/AppleJPEG
dyld: loaded: /System/Library/Frameworks/MetalPerformanceShaders.framework/Frameworks/MPSCore.framework/Versions/A/MPSCore
dyld: loaded: /System/Library/Frameworks/MetalPerformanceShaders.framework/Frameworks/MPSImage.framework/Versions/A/MPSImage
dyld: loaded: /System/Library/Frameworks/MetalPerformanceShaders.framework/Frameworks/MPSNeuralNetwork.framework/Versions/A/MPSNeuralNetwork
dyld: loaded: /System/Library/Frameworks/MetalPerformanceShaders.framework/Frameworks/MPSMatrix.framework/Versions/A/MPSMatrix
dyld: loaded: /System/Library/PrivateFrameworks/MetalTools.framework/Versions/A/MetalTools
dyld: loaded: /System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGLU.dylib
dyld: loaded: /System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGFXShared.dylib
dyld: loaded: /System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGL.dylib
dyld: loaded: /System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGLImage.dylib
dyld: loaded: /System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libCVMSPluginSupport.dylib
dyld: loaded: /System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libCoreVMClient.dylib
dyld: loaded: /usr/lib/libcups.2.dylib
dyld: loaded: /System/Library/Frameworks/Kerberos.framework/Versions/A/Kerberos
dyld: loaded: /System/Library/Frameworks/GSS.framework/Versions/A/GSS
dyld: loaded: /usr/lib/libresolv.9.dylib
dyld: loaded: /usr/lib/libiconv.2.dylib
dyld: loaded: /System/Library/PrivateFrameworks/Heimdal.framework/Versions/A/Heimdal
dyld: loaded: /usr/lib/libheimdal-asn1.dylib
dyld: loaded: /System/Library/Frameworks/OpenDirectory.framework/Versions/A/OpenDirectory
dyld: loaded: /System/Library/PrivateFrameworks/CommonAuth.framework/Versions/A/CommonAuth
dyld: loaded: /System/Library/Frameworks/OpenDirectory.framework/Versions/A/Frameworks/CFOpenDirectory.framework/Versions/A/CFOpenDirectory
dyld: loaded: /System/Library/Frameworks/SecurityFoundation.framework/Versions/A/SecurityFoundation
dyld: loaded: /System/Library/PrivateFrameworks/APFS.framework/Versions/A/APFS
dyld: loaded: /usr/lib/libutil.dylib
dyld: loaded: /System/Library/Frameworks/CoreAudio.framework/Versions/A/CoreAudio
dyld: loaded: /System/Library/Frameworks/AudioToolbox.framework/Versions/A/AudioToolbox
dyld: loaded: /System/Library/PrivateFrameworks/AppleSauce.framework/Versions/A/AppleSauce
dyld: loaded: /System/Library/PrivateFrameworks/LinguisticData.framework/Versions/A/LinguisticData
dyld: loaded: /usr/lib/libmarisa.dylib
dyld: loaded: /System/Library/PrivateFrameworks/Lexicon.framework/Versions/A/Lexicon
dyld: loaded: /usr/lib/libChineseTokenizer.dylib
dyld: loaded: /usr/lib/libcmph.dylib
dyld: loaded: /System/Library/PrivateFrameworks/LanguageModeling.framework/Versions/A/LanguageModeling
dyld: loaded: /System/Library/Frameworks/CoreData.framework/Versions/A/CoreData
dyld: loaded: /System/Library/PrivateFrameworks/CoreEmoji.framework/Versions/A/CoreEmoji
dyld: loaded: /System/Library/Frameworks/ServiceManagement.framework/Versions/A/ServiceManagement
dyld: loaded: /System/Library/PrivateFrameworks/BackgroundTaskManagement.framework/Versions/A/BackgroundTaskManagement
dyld: loaded: /usr/lib/libxslt.1.dylib
dyld: loaded: /Users/vike/.vscode/extensions/ms-dotnettools.csharp-2.4.4-darwin-x64/.debugger/x86_64/libcoreclr.dylib
dyld: loaded: /Users/vike/.vscode/extensions/ms-dotnettools.csharp-2.4.4-darwin-x64/.debugger/x86_64/libclrjit.dylib
dyld: loaded: /Users/vike/.vscode/extensions/ms-dotnettools.csharp-2.4.4-darwin-x64/.debugger/x86_64/libSystem.Native.dylib
Waiting for communication on port 4711...
>> accepted connection from client
dyld: loaded: /Users/vike/.vscode/extensions/ms-dotnettools.csharp-2.4.4-darwin-x64/.debugger/x86_64/libvsdbg.dylib
dyld: loaded: /Users/vike/.vscode/extensions/ms-dotnettools.csharp-2.4.4-darwin-x64/.debugger/x86_64/libvsdebugeng.impl.dylib
dyld: loaded: /Users/vike/.vscode/extensions/ms-dotnettools.csharp-2.4.4-darwin-x64/.debugger/x86_64/libdbgshim.dylib
>> client connection closed
```

</details>

**Comment by gregg-miskelly** [comment]
Thanks @vike2000. I guess the good news is that the correct `libdbgshim.dylib` is being used, but that kind of makes this problem even weirder since this seems to imply that the correct dylib file loaded, but was somehow the loader wasn't able to return one of its exports. Unfortunately, at least on my Mac, I wasn't able to get dyld to provide any further information. You can look at `man dyld` if you want to try any of the other switches in case they reveal more on your computer.

I few more questions:
1. Are you able to upgrade your version of macOS? Looking at the [.NET Support List](https://github.com/dotnet/core/blob/main/release-notes/7.0/supported-os.md#macos) 10.13 is out-of-support and I wonder if that is the cause since both you and the original poster are on 10.13.
2. Do you see this when debugging 'hello world'? Or is this specific to `godot`?

**Comment by vike2000** [comment]
Thanks for looking at it and your response @gregg-miskelly.

1. Unfortunately I'm unable to upgrade my macOS because of lacking hardware requirements, lacking in turn because of economics.
2. `$ dotnet new console`, with or without following `global.json`, after adding launch and tasks json "assets" from a notification, behaves the same as the `godot` project.
    <details><summary>`global.json`</summary>
    
    ``` json
    {
      "sdk": {
        "version": "6.0.100",
    	"rollForward": "feature"
      }
    }
    ```
    </details>
    
    I had forgotten how I solved debugging in another C# project (working non-trivial dotnet6 web with efcore and identity framework), but then in VS for Mac (Community) 2019 (not VSCode): I had made `Program.cs` do `Console.ReadKey()` and then debug `$ dotnet run` by name in Attach.
    
    Adding `Console.ReadKey()` to the Hello World project Attach in VS for Mac (Community) 2019 **works** while in VSCode again I get **Unknown Error: 0x92330062**

I really like VSCode much more than VS for Mac (Community) 2019. Also I'm not sure where/how/if I can wait to continue execution in a `godot` project with `Console.ReadKey()` like in a `Program.cs`.

I understand that my OS/hardware is ancient but since debugging csharp/dotnet6 *can* work on my OS I was hoping it could in VSCode too if I could just get everything to load/run correctly...VSCode too if I could just get everything to load/run correctly...

@gregg-miskelly:
> the loader wasn't able to return one of its exports

Can you hint on where/what the loader and relevant exports are here? I'm curious if I could look at hacking on the extension, though I doubt I'd be able to grasp the intricacies easily enough to be worth it on my own end.

Any way, thanks again.

**Comment by gregg-miskelly** [comment]
Unfortunately, the debugger and language service run on top of .NET 7 (and probably .NET 8 soon), so even if your app doesn't work on top of .NET 7, the debugger does. Also, the library that is failing to fully load (libdbgshim.dylib) is used regardless of .NET version. My **guess** is that it works in VS For Mac only because that is using an older version of the debugger that uses an older version of libdbgshim.dylib.

You could try making a program that loaded libdbgshim.dylib to try and discover what export is failing to load, but even if you know the answer it probably doesn't get you closer to a solution.

Here are the only work arounds I can think of given that you can't upgrade macOS:
1. Use Linux or Windows in a VM instead
2. Try building your own libdbgshim.dylib and replacing the version that vsdbg uses. Source code is in https://github.com/dotnet/diagnostics
3. Roll back to an older version of the C# extension and don't upgrade

**Comment by vike2000** [comment]
Yeah, I have been able to install/compile/run .NET 7 to some extent but only using versions like `7.0.100-preview.7.…` (prev. mentioned), and I don't think I'd be able to push that envelope further.

Thanks for your insights and possible routes in my scarce situation. Thinking about it I have been successful debugging DirectX C++ on Windows 10 in a Docker "devcontainer" in the past, so that could work, and though option 2 sounds intriguing I'd have to see if it'd be worth the pain :)

Opting for your idea 3 to work around I found that I get the original issue in the earliest major version 2 - 2.0.320 - while I can debug successfully with the (currently;) latest major version 1 - 1.26.0 -
so I'm now a happy camper over here!
